__tests__/app.test.js

```diff
@@ -35,57 +35,83 @@ describe("GET/api/topics", () => {
+  it("404: responds with an error", () => {
+    return request(app)
           .get('/not-a-route')
           .expect(404)
           .then(({ body }) => {
             expect(body.msg).toBe('Route not found');
           });
       });
-  });
-describe('GET/api/articles/:article_id', () => {
-    it('200: responds with requested article', () => {
+    });
+    describe("GET/api/articles/:article_id", () => {
+      it("200: responds with requested article", () => {
         const articleId = 1;
         return request(app)
         .get(`/api/articles/${articleId}`)
         .expect(200)
         .then((res) => {
-            const { body } = res;
-            const { article } = body;
-            expect(article).toHaveProperty('author', expect.any(String));
-            expect(article).toHaveProperty('title', expect.any(String));
-            expect(article).toHaveProperty('article_id', articleId);
-            expect(article).toHaveProperty('body', expect.any(String));
-            expect(article).toHaveProperty('topic', expect.any(String));
-            expect(article).toHaveProperty('created_at', expect.any(String));
-            expect(article).toHaveProperty('votes', expect.any(Number));
-            expect(article).toHaveProperty('article_img_url', expect.any(String));
-        });
+          const { body } = res;
+          const { article } = body;
+          expect(article).toEqual(expect.objectContaining({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: articleId,
+            body: expect.any(String),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+          }));
+          });
     });
-    it('404: responds with an error if article_id is not found', () => {
+    it("404: responds with an error if article_id is not found", () => {
       const article_id = 0;
       return request(app)
       .get(`/api/articles/${article_id}`)
       .expect(404)
       .then((res) => {
-          expect(res.body.msg).toBe(`Article ${article_id} not found`);
+        expect(res.body.msg).toBe(`Article ${article_id} not found`);
       });
     });
-    it('400: respond with an error if the id given is not a number', () => {
+    it("400: respond with an error if the id given is not a number", () => {
       return request(app)
       .get('/api/articles/article_id')
       .expect(400)
       .then(({body}) => {
-              expect(body.msg).toBe('Invalid id')
+        expect(body.msg).toBe('Invalid id')
       });
     });
   });
+  describe("GET/api/articles", () => {
+    it("200: responds with array of articles", () => {
+      return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        const { body } = res;
+        const articles = body;
+          expect(Array.isArray(articles)).toBe(true);
```

Comment:
Line 102 is implicitly proven when you use the .forEach method, as .forEach can only be used on an array. Feel free to remove this.

Reply:
resolved, thanks!